tests/tests_pytorch/loops/test_evaluation_loop.py

```diff
@@ -496,3 +496,28 @@ def test_step(self, batch, batch_idx):
     actual = trainer.num_val_batches if fn == "validate" else trainer.num_test_batches
     assert actual == [3, 2]
     assert seen == expected
+
+
+def test_evaluation_loop_when_batch_idx_argument_is_not_given(tmpdir):
+    class TestModel(BoringModel):
+        def __init__(self) -> None:
+            super().__init__()
+            self.validation_step_called = False
+            self.test_step_called = False
+
+        def validation_step(self, batch):
+            self.validation_step_called = True
+            return {"x": self.step(batch)}
+
+        def test_step(self, batch):
+            self.test_step_called = True
+            return {"y": self.step(batch)}
+
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=1)
```

Comment:
We should probably set `logger=False`, `enable_checkpointing=False`, `enable_progress_bar=False` in these tests

Reply:
> Revised, I feel the reason for doing this is to reduce the execution of unrelated code during testing. Are there any other reasons why we need to do this?

Reply:
That's right. It'll make the test run faster too